### PR TITLE
op-upgrade: introduce

### PR DESCRIPTION
**Pull request by tynes**
**Description**

CLI for building upgrade safe bundles. The docs can be found in the included `README`

### CODE REVIEW

File: op-chain-ops/cmd/op-upgrade/main.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"slices"
+
+	"golang.org/x/exp/maps"
+
+	//"strings"
```
Comment (tynes) [inline op-chain-ops/cmd/op-upgrade/main.go]: dead code

**Review by protolambda — APPROVED** [review]

**Review by refcell — APPROVED** [review]
Double checked the l1 upgrade calls and params, looks good to me.

**Review by refcell — APPROVED** [review]

File: op-chain-ops/cmd/op-upgrade/main.go (hunk 1 above)
Comment (refcell) [reply op-chain-ops/cmd/op-upgrade/main.go]: this comment is blocking mergify auto-merging fyi - not sure if this is intentional

**Comment by tynes** [comment]
@mergifyio refresh

**Pull request merged**